modules/statistics/php/statistics_site.class.inc

```diff
@@ -41,6 +41,10 @@ class Statistics_Site extends \NDB_Form
         $user     =& \User::singleton();
         $centerID = htmlspecialchars($_REQUEST['CenterID']);
         //TODO: Create a permission specific to statistics
+        if (is_null($centerID)) {
+            error_log("CenterID passed in statistics module is null.");
```

Comment:
shouldn't this be an exception instead than an error log ? isn't this an error

Reply:
one of two things can be happening here.
either a null object is an expected value (in which case just leave it be without an error) or it is never supposed to be null and then we need an exception. putting an error log without an exception will just add lines to the already crowded error log...

Reply:
`hasCenterPermission` returns true if the user has the permission and center. To determine the center, it calls `hasCenter` from the user class which returns `in_array($center_id, $this->getCenterIDs(), true);`. `getCenterIDs` never contains null values unless there is a null centerID in the DB. Given a null `$centerID`, the function in question returns false in its current state. John's PR removed warnings without changing functionality, my goal is to do the same thing. Here specifically we are cutting down a multi-line php Notice to a 1 line `error_log` call.

Reply:
`in_array(null, $arr)` only returns true if `$arr` contains `null`

Reply:
I'm with Rida that either null is expected, and there shouldn't be an error given, or it should never happen, and the code should throw an exception.. but to be honest, it's academic. The if statement is always false and this code is unreachable.

I'll leave why as an exercise to the reader.

Reply:
...... `htmlspecialchars(null)` doesn't return null, I'm guessing